build.gradle

```diff
@@ -9,16 +9,16 @@ plugins {
     id 'checkstyle'
 
     // community plugins
-    id 'com.palantir.docker' version '0.25.0'
-    id 'com.diffplug.spotless' version '5.11.0'
-    id 'net.ltgt.errorprone' version '1.3.0'
-    id 'com.github.spotbugs' version '4.6.0'
+    id 'com.palantir.docker' version '0.28.0'
+    id 'com.diffplug.spotless' version '5.15.0'
+    id 'net.ltgt.errorprone' version '2.0.2'
+    id 'com.github.spotbugs' version '4.7.5'
     id 'org.ajoberstar.grgit' version '4.1.0'
-    id 'com.github.ksoichiro.console.reporter' version '0.6.2'
-    id 'se.bjurr.violations.violations-gradle-plugin' version '1.48'
-    id 'com.github.johnrengelman.shadow' version '6.1.0'
-    id 'org.sonarqube' version '3.1.1'
-    id 'team.yi.semantic-gitlog' version '0.5.13'
+    id 'com.github.ksoichiro.console.reporter' version '0.6.3'
+    id 'se.bjurr.violations.violations-gradle-plugin' version '1.50.16'
+    id 'com.github.johnrengelman.shadow' version '7.0.0'
+    id 'org.sonarqube' version '3.3'
+    //id 'team.yi.semantic-gitlog' version '0.5.17' // isn't compatible with Gradle 7.2 yet
```

Comment:
At some point in time I updated everything, because I did assume it is a bug in the parallel execution of JUnit which was quite new. As JUnit parallel execution is build into Gradle I had to update gradle.

The plugin is only used for generating the changelog as far as I can tell. Executing the plugin with a lower Gradle is much less effort, than doing the whole Gradle update again. Thus I would argue we update it, as soon as it is available for Gradle 7. If the plugin will not be updated, we anyways have to check for a different plugin, as Gradle 8 is already around the corner and the Gradle 7 update would have been due anyways.

Reply:
I agree, haven't thought about running the changelog task with an older version. This makes sense.but how can we achieve this?

Reply:
simply change the wrapper version and run the task, just work just fine :)